WordPress/Classes/ViewRelated/Site Creation/FinalAssembly/SiteAssemblyContentView.swift

```diff
@@ -0,0 +1,282 @@
+
+import UIKit
+
+// MARK: SiteAssemblyContentView
+
+/// This view is intended for use as the root view of `SiteAssemblyWizardContent`.
+/// It manages the state transitions that occur as a site is assembled via remote service dialogue.
+///
+class SiteAssemblyContentView: UIView {
+
+    // MARK: Properties
+
+    private struct Parameters {
+        static let animationDuration                            = TimeInterval(0.25)
+        static let assembledSiteScaleFactorWidth                = CGFloat(0.79)
+        static let assembledSiteScaleFactorHeight               = CGFloat(0.79)
+        static let buttonContainerScaleFactor                   = CGFloat(2)
+        static let completionLabelHorizontalInset               = CGFloat(37)
+        static let labelSpacingScaleFactor                      = CGFloat(0.032)
+        static let statusStackViewSpacing                       = CGFloat(16)
+    }
+
+    private var completionLabelTopConstraint: NSLayoutConstraint?
+
+    private(set) var completionLabel: UILabel
+
+    private let statusLabel: UILabel
+
+    private let activityIndicator: UIActivityIndicatorView
+
+    private(set) var statusStackView: UIStackView
+
+    private var assembledSiteTopConstraint: NSLayoutConstraint?
+
+    private(set) var assembledSiteView: AssembledSiteView?
+
+    private var buttonContainerBottomConstraint: NSLayoutConstraint?
+
+    private var buttonContainerContainer: UIView?
+
+    var buttonContainerView: UIView? {
+        didSet {
+            installButtonContainerView()
+        }
+    }
+
+    var domainName: String? {
+        didSet {
+            installAssembledSiteView()
+        }
+    }
+
+    var status: SiteAssemblyStatus = .idle {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    // MARK: SiteAssemblyContentView
+
+    init() {
+        self.completionLabel = {
```

Comment:
I don't see this is as a blocker, but I find this way of creating the properties makes the initialiser difficult to parse, as it turns into a long method with several long blocks of indented code. Again, my personal preference, and not a blocker, but would you consider declaring these properties as lazy var? (I still like a lot the idea of declaring in a closure)